addons/binding/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/handler/SerialOceanicThingHandler.java

```diff
@@ -0,0 +1,315 @@
+
+                    if ((len = inputStream.read(tmpData)) > -1) {
+                        if (logger.isTraceEnabled()) {
+                            StringBuilder sb = new StringBuilder();
+                            for (int i = 0; i < len; i++) {
+                                sb.append(String.format("%02X", tmpData[i]));
+                                sb.append(" ");
+                            }
+                            logger.trace("Read {} bytes : {}", len, sb.toString());
+                        }
+                        for (int i = 0; i < len; i++) {
+                            if (logger.isTraceEnabled()) {
+                                logger.trace("Byte {} equals '{}' (hex '{}')", i, new String(new byte[] { tmpData[i] }),
+                                        String.format("%02X", tmpData[i]));
+                            }
+
+                            if (tmpData[i] != lineFeed && tmpData[i] != carriageReturn && tmpData[i] != nullChar) {
+                                dataBuffer[index++] = tmpData[i];
+                                if (logger.isTraceEnabled()) {
+                                    logger.trace("dataBuffer[{}] set to '{}'(hex '{}')", index - 1,
+                                            new String(new byte[] { dataBuffer[index - 1] }),
+                                            String.format("%02X", dataBuffer[index - 1]));
+                                }
+                            }
+
+                            if (i > 0 && (tmpData[i] == lineFeed || tmpData[i] == carriageReturn
+                                    || tmpData[i] == nullChar)) {
+                                if (index > 0) {
+                                    if (logger.isTraceEnabled()) {
+                                        logger.trace("The resulting line is '{}'",
+                                                new String(Arrays.copyOf(dataBuffer, index)));
+                                    }
+                                    line = StringUtils.chomp(new String(Arrays.copyOf(dataBuffer, index)));
+                                    line = line.replace(",", ".");
+                                    line = line.trim();
+                                    index = 0;
+
+                                    lastLineReceived = line;
+                                    break;
+                                }
+                            }
+
+                            if (index == bufferSize) {
+                                index = 0;
+                            }
+                        }
+                    }
+
+                    try {
+                        Thread.sleep(sleep);
+                    } catch (InterruptedException e) {
+                        // Move on silently
+                    }
+                }
+            } catch (Exception e) {
+                logger.error("An exception occurred while reading serial port  : {}", e.getMessage(), e);
```

Comment:
Should the thing status be updated should something be retried?

Reply:
@martinvw Github reports this one as open, but was already fixed.